docker/Dockerfile

```diff
@@ -1,101 +1,125 @@
-FROM ghcr.io/pulp-platform/snitch_cluster@sha256:9978f9d0e3e6b7ad1b2c1f0eb628a113226e8443bc42ae8b4ddd3f34d9528e96 as builder
-
-RUN git clone --recursive https://github.com/nazavode/snitch_cluster.git /src \
- && cd /src \
- && git reset --hard 6c63f4c65d115ce16193f4f9a030ac08416a4acf \
- && git submodule update \
- # verilator model
- && cd /src/target/snitch_cluster \
- && make CFG_OVERRIDE=cfg/fdiv.hjson bin/snitch_cluster.vlt \
- # spike-dasm
- && cd /src/target/snitch_cluster/work-vlt/riscv-isa-sim \
- && ./configure --prefix=/opt/snitch-spike \
- && make install \
- # snitch runtime
- && cd /src/target/snitch_cluster \
- && make DEBUG=ON sw \
- # clang+llvm+lld
- && mkdir -p /opt/snitch-llvm \
- && wget -qO- https://github.com/pulp-platform/llvm-project/releases/download/0.12.0/riscv32-pulp-llvm-ubuntu1804-0.12.0.tar.gz | \
-    tar xz --strip-components=1 -C /opt/snitch-llvm \
- # python 3.11
- && apt update -y \
- && apt install -y \
-      build-essential \
-      zlib1g-dev \
-      libncurses5-dev \
-      libgdbm-dev \
-      libnss3-dev \
-      libssl-dev \
-      libreadline-dev \
-      libffi-dev \
-      libsqlite3-dev \
-      wget \
-      libbz2-dev \
- && mkdir -p /tmp/python-src \
- && wget -qO- https://www.python.org/ftp/python/3.11.5/Python-3.11.5.tgz | \
-    tar xz --strip-components=1 -C /tmp/python-src \
- && cd /tmp/python-src \
- && ./configure --prefix=/opt/python3.11 \
- && make install
-
-FROM ubuntu:18.04 as toolchain
+#-------------------------------------------------------------------------------
+# Stage 1: Builder
+# Clones the repository and builds the necessary artifacts.
+#-------------------------------------------------------------------------------
+FROM ghcr.io/pulp-platform/snitch_cluster:main AS builder
+
+# Clone the snitch_cluster repository into /repo
+RUN git clone -b fix-deps https://github.com/pulp-platform/snitch_cluster /repo
+
+WORKDIR /repo
+
+# Patch .gitmodules to replace SSH URLs with HTTPS
+RUN sed -i 's|git@github.com:|https://github.com/|g' .gitmodules
+
+# Initialize and update submodules (now using HTTPS)
+RUN git submodule update --init --recursive
+
+RUN make -C target/snitch_cluster verilator
+
+RUN make -C target/snitch_cluster DEBUG=ON sw
+
+WORKDIR /
+
+RUN cd /repo/target/snitch_cluster/work-vlt/riscv-isa-sim \
+  && ./configure --prefix=/opt/snitch-spike \
+  && make install
+
+WORKDIR /src
+
+RUN curl -L -o riscv32-snitch-llvm-ubuntu2204-15.0.0-snitch-0.2.0.tar.gz https://github.com/pulp-platform/llvm-project/releases/download/15.0.0-snitch-0.2.0/riscv32-snitch-llvm-ubuntu2204-15.0.0-snitch-0.2.0.tar.gz
+
+RUN tar -xzf riscv32-snitch-llvm-ubuntu2204-15.0.0-snitch-0.2.0.tar.gz -C /src
```

Comment:
This used to be `snitch-llvm`, we need to see what this breaks. I'm leaning towards leaving the above name as is as it has all the information from upstream